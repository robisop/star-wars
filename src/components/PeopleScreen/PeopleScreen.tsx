import {ListView} from 'components/common/ListView';
import {usePeople} from 'hooks/usePeople';

export default function PeopleScreen() {
  const {
    data,
    isLoading,
    isError,
    fetchNextPage,
    hasNextPage,
    isFetchingNextPage,
  } = usePeople();

  return (
    <ListView
      data={data?.pages.map(page => page.results).flat()}
      hasNextPage={hasNextPage}
      isLoading={isLoading}
      isError={isError}
      isFetchingNextPage={isFetchingNextPage}
      fetchNextPage={fetchNextPage}
    />
  );
}
