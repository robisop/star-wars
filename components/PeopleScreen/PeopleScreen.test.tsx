import {render} from '@testing-library/react-native';
import renderer, {ReactTestRendererJSON} from 'react-test-renderer';

import PeopleScreen from './PeopleScreen';

describe('<PeopleScreen />', () => {
  it('has 1 child', () => {
    const tree = renderer
      .create(<PeopleScreen />)
      .toJSON() as ReactTestRendererJSON;
    expect(tree.children?.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<PeopleScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Hello World message on the home page', async () => {
    const screen = render(<PeopleScreen />);
    expect(screen.getByText('People!')).toBeDefined();
  });
});
