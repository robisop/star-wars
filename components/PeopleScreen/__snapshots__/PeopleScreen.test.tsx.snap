// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeopleScreen /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    People!
  </Text>
</View>
`;
